### ISSUE TITLE
Hatch Rest "White Color"

### ISSUE THREAD

**Issue by jesse-schein** [dahlb/hatch_rest_api#24]
Have you considered to make the Hatch Rest API call to send "w" true or whatever value it expects in this code segment?

``` def set_color(self, red: int, green: int, blue: int, brightness: int, random: bool = False):
        self._update(
            {
                "c": {
                    "r": convert_from_hex(red),
                    "g": convert_from_hex(green),
                    "b": convert_from_hex(blue),
                    "i": convert_from_percentage(brightness),
                    "W": False,
                    "R": random,
                }
            }
        )
```

The Hatch App sets white to a different color then when home assistant sends white and I assume its because the hatch API is getting 255 255 255 instead of white true and it displays differently on the hatch because the LEDs are probably RGB and a white channel perhaps. The white channel makes the "white" look nicer, despite being warmer in color like 2700k or 2500k.

Maybe in this section of code we could detect if 255 255 255 is sent and then set W to true?

Thoughts?

**Comment by dahlb** [comment]
seems better to handle this in the home assistant specific code so I tried it https://github.com/dahlb/ha_hatch/commit/1d6b67a550a1bfef63ad3b164b4bb82675e1549c
it's released in ha_hatch 1.21.0 let me know if that works well for you. I generally utilize routines so I don't use these code paths but I like your suggestion

**Comment by jesse-schein** [comment]
Thanks so much for the fast work!

I tested it and it works great. 

If anyone ever digs this up, the default home assistant “white” isn’t true white as in 255 255 255. It probably is doing some scaling since green is actually a brighter led normally so it’s scaled down to “look white”. 

We probably could have made it so hatch rest plus supports kelvin support since it appears to have a dedicated white channel and rgb channel.
